README.md

```diff
@@ -106,18 +108,39 @@ Event loop does not say much about when events are dispatched:
 3. Microtasks get executed immediately after a task.
 
 4. Render part of the loop gets executed on vSync, and delivers events in the following order:
+    1. dispatch pending UI events
+
+    2. 'resize' event
+
+    3. 'scroll' event
+
+    4. mediaquery listeners
+
+    5. 'CSSAnimation' events
 
-    1. 'resize' event
+    6. Observers
 
-    2. 'scroll' event
+    7. rAF
 
-    3. mediaquery listeners
+## Pending UI events
 
-    4. 'CSSAnimation' events
+There are two classes of UI events:
 
-    5. Observers
+1. Discrete - those that aren't continuous (eg. mousedown, mouseup,
+   touchstart, touchend)
 
-    6. rAF
+2. Continuous - mousewheel, wheel, mousemove, pointermove, touchmove.
+
+Continuous events may be coalesced (updating positions, magnitude) with
+matching events (that haven't yet been dispatched) while being held in the
+UI event task queue.
+
+Discrete events should be dispatched right as soon as possible when received
+from the hardware. Continuous events can be held and dispatched in the
+render part of the event loop. Ordering of discrete and continuous events
+must be preserved. If a discrete event is received while a continuous event
```

Comment:
I get slightly confused by this sentence, who should run right away?
"If a discrete event is received while a continuous event is being held in the queue for the next vSync signal it should run right away to prevent the discrete event from being delayed."

What do you think about this wording:
"If a discrete event is received, all continuous events in the task queue should run immediately to prevent discrete event from being delayed."